core/src/main/java/org/mule/runtime/core/internal/event/DefaultEventContext.java

```diff
@@ -167,7 +167,7 @@ public BaseEventContext getRootContext() {
    */
   public DefaultEventContext(FlowConstruct flow, ComponentLocation location, String correlationId,
                              Optional<CompletableFuture<Void>> externalCompletion) {
-    this(flow, flow.getExceptionListener(), location, correlationId, externalCompletion);
+    this(flow, NullExceptionHandler.getInstance(), location, correlationId, externalCompletion);
```

Comment:
is there any place in which a non null exception handler is set? If the answer is no, then simply remove the concept from here.

Reply:
good catch, more cleanup on the way